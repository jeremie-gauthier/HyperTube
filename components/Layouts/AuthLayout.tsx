import { useResponsiveAttribute } from "@/hooks/useSelector";
import Image from "next/image";
import { FlexCol } from "../Flex";
import styles from "./AuthLayout.module.scss";

type LayoutProps = {
  children: React.ReactNode;
};

export default function Layout({ children }: LayoutProps): JSX.Element {
  const isTabletOrMobile = useResponsiveAttribute();

  return isTabletOrMobile ? (
    <MobileView>{children}</MobileView>
  ) : (
    <DesktopView>{children}</DesktopView>
  );
}

const MobileView = ({ children }: LayoutProps) => (
  <FlexCol className={styles.layoutMobile}>
    <div className="mt-4 ml-3">
      <Image
        layout="intrinsic"
        height={32}
        width={118}
        src="/icons/hypertube.png"
        alt="Hypertube logo"
        priority
      />
    </div>
    {children}
  </FlexCol>
);

const DesktopView = ({ children }: LayoutProps) => (
  <FlexCol className={styles.layoutDesktop} data-testid="layout-div">
    <Image
      layout="fill"
      src="/img/landing.jpg"
      alt="Movies"
      priority
      quality={100}
    />
    <div className={styles.brandLogo}>
      <Image
        layout="intrinsic"
        height={54}
        width={200}
        src="/icons/hypertube.png"
        alt="Hypertube logo"
        priority
      />
    </div>
    <FlexCol className={styles.container}>{children}</FlexCol>
  </FlexCol>
);
