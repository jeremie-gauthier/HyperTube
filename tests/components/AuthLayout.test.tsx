import React from "react";
import AuthLayout from "@/components/Layouts/AuthLayout";
import renderer from "react-test-renderer";

test("<AuthLayout /> should render correctly", () => {
  const tree = renderer
    .create(
      <AuthLayout>
        <div />
      </AuthLayout>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
