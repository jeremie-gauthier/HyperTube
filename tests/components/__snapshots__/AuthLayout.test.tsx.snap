// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLayout /> should render correctly 1`] = `
<div
  className="flexCol layout"
  data-testid="layout-div"
>
  <div
    className="brandLogo"
  />
  <div
    className="flexCol container"
  >
    <div />
  </div>
</div>
`;
